app/services/hyrax/working_directory.rb

```diff
@@ -3,23 +3,6 @@ module Hyrax
   # @deprecated Use JobIoWrapper instead
   class WorkingDirectory
     class << self
-      # Returns the file passed as filepath if that file exists. Otherwise it grabs the file from repository,
-      # puts it on the disk and returns that path.
-      # @param [String] repository_file_id identifier for Hydra::PCDM::File
-      # @param [String] id the identifier of the FileSet
-      # @param [String, NilClass] filepath path to existing cached copy of the file
-      # @return [String] path of the working file
-      def find_or_retrieve(repository_file_id, id, filepath = nil)
```

Comment:
let's deprecate this, instead of removing. it's public API now, so this would avoid the need to bump version to 4.0.0.